### DISCUSSION TITLE
STM32 : Micropython as a Task in FreeRTOS

### DISCUSSION THREAD

**Discussion by bhavin03** [micropython/micropython#15808] [STM32 / Pyboard]
Hi, 
I have tried to implement Micropython as a Task in FreeRTOS on a NUCLEO-F767ZI Board. It compiles and run fine until I execute any Python Line in REPL mode (e.g. print('hello world')). During byte code execution it's getting HardFaults during context switch in xTaskIncrementTick() in Systick handler . What I have basically done is that, I started with a bare minimum project and implemented FreeRTOS using STMCubeMX. It looks like stack is getting corrupted during context switching. Is there anything to take care of when using Micropython as FreeRTOS  Task ?? Micropython runs fine in super loop, but getting problems when using with RTOS. I have looked ESP32 and CC3200 Ports, but couldn't find any clue there.. 

My Project do need RTOS implementation and Micropython should run as a separate Task.

Thank you..

**Comment by MHe999** [comment]
Hi @bhavin03, 
did you get this solved? I'm attempting the exact same thing (STM32+FreeRTOS+MicroPython), but don't even have a compiling project yet. If there are some tricks to do with the FreeRTOS to get it really working, I would be very interested.

And may I ask, how did you build and include the MicroPython to the Cube project?

Thanks.
